sdk/core/azure-core-v2/src/main/java/com/azure/v2/core/implementation/http/policy/AuthorizationChallengeParser.java

```diff
@@ -0,0 +1,121 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.v2.core.implementation.http.policy;
+
+import com.azure.v2.core.utils.CoreUtils;
+import io.clientcore.core.http.models.HttpHeaderName;
+import io.clientcore.core.http.models.Response;
+import io.clientcore.core.implementation.http.HttpResponse;
+
+/**
+ * Parses Authorization challenges from the {@link HttpResponse}.
+ */
+public final class AuthorizationChallengeParser {
```

Comment:
Please use the utility parser available in `AuthUtils`. It is a strict implementation of the `WWW-Authenticate` spec.